### ISSUE TITLE
decimal numerals in words.txt

### ISSUE THREAD

**Issue by pgmmpk** [slavonic/cu-tex#5]
@typiconman  It is a mistake?

See line 651 and after in words.txt (unnormalized):

```
  650 бдѣ́н-на-гѡ
  651 без-2бо́ж-на-гѡ
  652 без-2бо́ж-но-мꙋ
  653 без-2бо́ж-ныхъ
  654 без-2бѡ́ж-ны-ѧ
  655 без-2бо-лѣ́з-нен-нѡ
  656 без-2бѣ́д-но-е
  657 без-2ви-но́в-на-гѡ
  658 без-2грѣ́ш-на-го
  659 без-2грѣ́ш-не
```

also

```
  973 бла-го-4вѣ-ство-ва́-нї-е
  974 бла-го-4вѣ-ство-ва́-нї-е
  975 бла-го-4вѣ-ство-ва́-ше
  976 бла-го-4вѣ-ствꙋ́-ю-ще
  977 бла-го-4вѣ-ствꙋ́-ю-щей
  978 бла-го-4вѣ-ствꙋ́-ю-щей
  979 бла-го-4вѣ-ствꙋ́-ѧй
  980 бла-го-4вѣ-ствꙋ́-ѧй
  981 бла-го-4вѣ-стѝ
  982 бла-го-4вѣ-стѝ
  983 бла-го-4вѣ-сти́-ти
  984 бла-го-4вѣ-ща́-ше
```

**Comment by typiconman** [comment]
According to the OPatGen documentation,

> The words in our dictionary can be _weighted_. If there is a number at any interletter position in the input data, the position is counted as many times as the number says. It brings the possiblity ot weight some words in our dictionary more heavily ... For example hy-2phen-a3-ti7on means that the position between y and p will be counted as it appeared twice and the position between a and t as it appeared three times.

My idea was to try to account for the fact that in compound words we prefer the hyphen to occur between the two parts, so, e.g.,
благо- вѣствꙋй is pereferrable to бла- говѣствꙋй.
Whether or not this actually has any bearing on the results -- I do not know, and I haven't tested.

**Comment by pgmmpk** [comment]
Cool, did not know that OPatGen supports that. Liang mentioned in his thesis that he used word
weighting to get better patterns, but did not include this feature in his `patgen`.

I'm pretty sure that hyphenation algorithm does not have the concept of "weighting".
My understanding is that these numbers are used for training only -- to control the "badness" 
and "goodness" of hyphen prediction. Basically, in the famous formula:

```
         num_good * good_weight - num_bad *bad_weight <= threshold
```

`num_good` is now weighted number of predicted hyphens, while `num_bad` is now weighted
number of mispredicted hyphens. 

In other words, if pattern correctly selects a hyphen that is
weighted with 4: `abc-4def`, it is rewarded as if it predicted 4 correct hyphens. Conversely, if
it mispredicted a hyphen at a place weighted 5: `abc5def`, it is punished as if it falsely predicted
5 hyphens.

Thus, weighting should produce better patterns, but this matter only during the training. For благовѣствꙋй we will likely find patterns that cover all hyphenation points and therefore TeX will
see бла-го-вѣствꙋй and choose the one basing on line geometry.

Would be interesting to see how it plays in practice. I need to add this feature to `pypatgen`...

**Comment by typiconman** [comment]
Well бла- говѣствꙋй wouldn't be completely wrong. But if I had the choice between бла- говѣствꙋй and благо- вѣствꙋй when I got to the end of a line, I should, ideally, select благо- вѣствꙋй. In practice, I don't think this is possible in TeX. It would require something more complicated than just patterns.
